Quotebot

Eagle1456 opened the pull request:
Add quotebot functionality to Automoji.

Eagle1456 committed:
Start writing methods for quotebot

Eagle1456 committed:
Add main functionality to quotebot

Eagle1456 committed:
Add main functionality to quotebot

Eagle1456 committed:
Fix indentation and turn quotesChannel and quotes into dictionaries

AnthonyN1 commented:
Please resolve conflicts by removing any commits not related to quotebot functionality.

Eagle1456 committed:
Merge branch 'main' into quote

Eagle1456 committed:
Remove is_quote method and add styling edits

Eagle1456 committed:
Merge branch 'quote' of https://github.com/Eagle1456/Automoji into quote

Eagle1456 committed:
Merge branch 'main' into quote

Eagle1456 committed:
Add changes to quotebot

Eagle1456 committed:
Merge branch 'main' into quote

Eagle1456 committed:
fix quotebot changes

Eagle1456 commented:
Ok, should be in line now with the revival changes.

AnthonyN1 committed:
Remove references to custom_send

See commit 6816d6b

AnthonyN1 committed:
Remove unnecessary function annotations

See commit 6816d6b

AnthonyN1 committed:
Clean up the error functions

Replace print statments with the logger.
Let some explicitly caught errors go to the logger.

AnthonyN1 committed:
Make changes to set_quote_channel

AnthonyN1 committed:
Make changes to the rest of quotes.py

AnthonyN1 committed:
Remove add_quote (unused)

AnthonyN1 committed:
Add comments to the quotes dictionaries

AnthonyN1 commented:
Commited some changes to conform to the post-revival codebase. For future PRs, I will have an updated contributions.md that will hopefully cover all the code styling that Black doesn't.

Some features that I would be interested in seeing:
- It seems that quotes the users add after the quotes channel is added aren't processed. How can the bot actively add to the list?
- The two dictionaries are a little confusing to read, code-wise. Maybe they can be condensed into one dictionary, where the values are tuples (Channel, List)?

Everything else is good. Will merge with main.
